suite-native/module-trading/src/utils/buyFormValidationSchema.ts

```diff
@@ -0,0 +1,94 @@
+import { yup } from '@suite-common/validators';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+
+import { TradingBuyFormContext } from '../types';
+
+const getAmountLimitContext = ({ options }: yup.TestContext<unknown>) =>
+    options.context as TradingBuyFormContext;
+
+export const buyFormValidationSchema = yup.object({
+    cryptoValue: yup
+        .number()
+        .positive()
+        .test('crypto-min', (value, testContext) => {
+            const { currency, minCrypto, translate, CryptoAmountFormatter } =
+                getAmountLimitContext(testContext);
+
+            if (value === undefined || minCrypto === undefined) {
+                return true;
+            }
+
+            if (value >= parseFloat(minCrypto)) {
+                return true;
+            }
+
+            const min = CryptoAmountFormatter.format(minCrypto, {
+                symbol: currency.toLowerCase() as NetworkSymbol,
+                isBalance: true,
+            });
+
+            return testContext.createError({
+                message: translate('moduleTrading.validators.min', { min }),
+            });
+        })
+        .test('crypto-max', (value, testContext) => {
+            const { currency, maxCrypto, translate, CryptoAmountFormatter } =
+                getAmountLimitContext(testContext);
+
+            if (value === undefined || maxCrypto === undefined) {
+                return true;
+            }
+
+            if (value <= parseFloat(maxCrypto)) {
+                return true;
+            }
+
+            const max = CryptoAmountFormatter.format(maxCrypto, {
+                symbol: currency.toLowerCase() as NetworkSymbol,
+                isBalance: true,
+            });
+
+            return testContext.createError({
+                message: translate('moduleTrading.validators.max', { max }),
+            });
+        }),
+    fiatValue: yup
+        .number()
+        .positive()
+        .test('fiat-min', (value, testContext) => {
+            const { currency, minFiat, translate, FiatAmountFormatter } =
+                getAmountLimitContext(testContext);
+
+            if (value === undefined || minFiat === undefined) {
+                return true;
+            }
+
+            if (value >= parseFloat(minFiat)) {
+                return true;
+            }
+
+            const min = FiatAmountFormatter.format(minFiat, { currency });
+
+            return testContext.createError({
+                message: translate('moduleTrading.validators.min', { min }),
+            });
+        })
+        .test('crypto-max', (value, testContext) => {
+            const { currency, maxFiat, translate, FiatAmountFormatter } =
+                getAmountLimitContext(testContext);
+
+            if (value === undefined || maxFiat === undefined) {
```

Comment:
you can probably combine these two ifs into one
```suggestion
            if (value === undefined || maxFiat === undefined || value <= parseFloat(maxFiat)) {
```

Reply:
I am too old to use long logic expressions (when I was young I was fan of complex ones). Updated.